bootstrap: put is_building_snapshot state in IsolateData and introduce ERR_NOT_SUPPORTED_IN_SNAPSHOT (#47887)
nodejs/node: doc/api/errors.md

An attempt was made to use operations that can only be used when building
V8 startup snapshot even though Node.js isn't building one.

<a id="ERR_NOT_SUPPORTED_IN_SNAPSHOT"></a>

> ### `ERR_NOT_SUPPORTED_IN_SNAPSHOT`

Reviewer: What about using a more generic `ERR_NOT_SUPPORTED` error? The concrete description could be part of the message. We have many very specific error codes that likely do not help users all that much.

Author: I don't think a generic `ERR_NOT_SUPPORTED` would be quite useful in this case. What would one do after checking `error.code === ERR_NOT_SUPPORTED`? I can imagine the embedder put a TryCatch around e.g. LoadEnvironment and check for `error.code === ERR_NOT_SUPPORTED_IN_SNAPSHOT`, if that's true, simply give up creating a snapshot (as some kind of optional optimization) and then only do something with the raw code, otherwise simply treat it as a proper error that it cannot handle and fail. `ERR_NOT_SUPPORTED` would not be enough in this case, and then they'd end up..parsing the error message.

Author: Ping @BridgeAR does that answer the question?

> An attempt was made to perform operations that are not supported in the
> snapshot.

Reviewer 2: ```suggestion
An attempt was made to perform operations that are not supported when
building a startup snapshot.
```
